lib/standards/tests/contentMustFollowHeadings.js

```diff
@@ -15,11 +15,17 @@ module.exports = {
       ).map(function () { return this.nextSibling })[0]
     }
 
+    function nodeHasContent ($node) {
+      return $node.text().trim().length > 0
+    }
+
     $(headingSelector).each(function (index, heading) {
       var nextNode = heading.nextSibling || ancestorsNextSibling(heading)
       while (nextNode) {
-        if ($(nextNode).is(heading.tagName)) { break }
-        if ($(nextNode).text().trim().length > 0) { return }
+        var $nextNode = $(nextNode)
+        var $nextIframeNodesBody = $(nextNode).find('iframe:visible').contents().find('body')
```

Comment:
This will find an iframe _wherever_ it is inside the `<div>`, so would this case pass when it should actually fail?

```
<h1>Main heading</h1>
<div>
  <h2>This heading should fail the test</h2>
  <iframe src="perfect.html">...but this iframe will be found to have contents, therefore pass the test</iframe>
</div>
```

Reply:
Why should this fail?

Reply:
Isn't this scenario about testing that there is content after each heading? But in my example code above, we have a h1 immediately followed by a h2 without any content. Or have I misunderstood?

Reply:
Another heading still counts as "content". See lines 176-185 for an example.

Reply:
Oh ok, the "no heading immediately after another heading" rule only applies to headings _of the same level_, I guess?

https://github.com/andymsuk/bbc-a11y/blob/676755ea552dab1f3b9d898d504f24ae064295f3/features/standards/mag/structure/02_headings.feature#L194-L217

* h1 next to h2 passes
* h2 next to h2 fails

...right?

(Just trying to understand the difference.)

Also, ok, let's take my code and change it a bit:

```
<h1>Main heading</h1>
<div>
  <h1>This heading should fail the test</h1>
  <iframe src="perfect.html">...but this iframe will be found to have contents, therefore pass the test</iframe>
</div>
```

...^^ I think this PR might now make this code pass when it shouldn't, because it'll find an iframe and check that?

Reply:
This particular standard will pass, because there is content after each of the `h1` elements.
However, the standard for `Exactly one main heading` will fail, as there are now two `h1` elements.

If you change it to this though...
```
      <h1>Main heading</h1>
      <h2>This is a sub-heading</h2>
      <h2>This heading should fail the test</h2>
      <iframe src="http://localhost:54321/perfect.html"></iframe>
```
it does also fail because there's a `h2` after the `h2`.

__However__, if you change it to this...
```
      <h1>Main heading</h1>
      <h2>This is a sub-heading</h2>
      <div>
        <h2>This heading should fail the test</h2>
        <iframe src="http://localhost:54321/perfect.html"></iframe>
      </div>
```
Then it does incorrectly pass 😢 

You're great at finding these edge cases!!

*back to the drawing board*